### ISSUE TITLE
Many transactions pending in Metamask, but none came through to Etherscan

### ISSUE THREAD

**Issue by Guestquiver** [MetaMask/metamask-extension#3756]
Hi there,

I'm sorry but I'm new to Github. I couldn't find anything on Google so I came here.

TL;DR - My metamask now has a lot of pending transactions that have not created a transaction hash on Etherscan, neither after retrying with higher Gas prices (I used a high gas price already in the first place)

- I used Metamask many many times, but am not an experienced developer or whatsoever
- I use loose accounts on Metamask
- I use Brave browser, and cannot find any way to reset Metamask. Since I use loose accounts this would be an easy option
- I tried logging in and out
- I tried reinstalling Brave
- I assume using Metamask with Chrome and importing my accounts would solve the issue, but I do not want to use Chrome. I would like to use Brave, but I cannot find how to reset Metamask.
- I am unsure what to do now. Obviously I don't want all these transactions to go through, or create a Ghost account. Since the accounts are loose I am not dependent on Metamask itself, but this issue I think is potentially harmful for anyone.

So, how the problem came to existence (I think):

I used metamask with two DEX'es. Then I created a withdraw. A bit later I created a deposit on another DEX. I might have entered a wrong amount of eth to withdraw/deposit, so it might have been that I did not have the right amount of ether on the accounts.

Following:

- None of the transactions created a TX hash on Etherscan
- Trying to raise the gas price for every transaction did not do anything
- Creating a new transaction through MEW did not do anything ether
- I now have 10 transactions or so 'pending' on Metamask, without having a transaction ID

Question:

- Can I just reset Metamask internally? Or will I get issues with this? I've read something about nonce, and don't want to screw up more...

Maybe I'm missing an obvious solution here. Either way I think it's worth a shot.

Thank you,

GQ

**Comment by Guestquiver** [comment]
Update:

So I loaded it succesfully on my other system. No issues, no weird transactions, and I could just use MM and the Loose account again.

I'll be trying to reset this and see what happens.

**Comment by danfinlay** [comment]
Sorry for the slow reply:

My guess is that your "next transaction" was the one with a low gas price, blocking the rest of them. I would recommend increasing the gas price on that one.

Resetting the account could result in a ghost account, so this could be dangerous:
https://metamask.helpscoutdocs.com/article/37-ghost-accounts-ether-sending-away-whenever-funding-an-account

**Comment by danfinlay** [comment]
Closing this because it's not an actionable bug/feature goal. If you have questions about using MetaMask, please email support@metamask.io

**Comment by CruzMolina** [comment]
@danfinlay this is definitely related to #4022.

I believe it is MM-specific, because as @Guestquiver mentioned, when you import the mnemonic for the "clogged" wallet into another browser or even the same browser on another system, all of a sudden the "localized mempool" clears and every single tx that was pending goes through.

This is a huge UX problem. I notice you guys are working on it, so godspeed. It is a much needed fix.

(For reference, @Guestquiver had the issue on Brave and I've personally had the issue on Firefox).

**Comment by BrianSipple** [comment]
I just ran in to the same issue -- exactly as the OP described. 

I took it one step further, though, and attempted to cancel my pending transactions. I now have a large queue of both the original pending transactions, and pending cancellation requests that also appear to be going nowhere:

<img width="362" alt="screen shot 2018-10-19 at 11 44 59 pm" src="https://user-images.githubusercontent.com/5483853/47250949-3b4cb380-d3f9-11e8-9775-6dfaaab95184.png">

@danfinlay Even though this might not be an actionable issue, are there any notes we can leave here, for reference, regarding a solution? 

I feel at a complete loss for how to proceed -- or what might have caused the issue in the first place.

**Comment by bdresser** [comment]
@CruzMolina @BrianSipple we're tracking this issue in 4.16 here: https://github.com/MetaMask/metamask-extension/issues/5551

**Comment by REPTILEHAUS** [comment]
+1 on this issue, happened to me yesterday

**Comment by mscheele** [comment]
+1 (keep up the great work) :-)

**Comment by bdresser** [comment]
similar issue to this one was reported in https://github.com/MetaMask/metamask-extension/issues/4343, fixed in https://github.com/MetaMask/metamask-extension/pull/5752, should be out in the next 24hr

**Comment by joshricha** [comment]
Same here. I have the issue

**Comment by DiMeneses** [comment]
Same, anybody solved it?

**Comment by bdresser** [comment]
@joshricha @DiMeneses and anyone reporting this recently - check on the status of the transaction with the lowest nonce. Has it appeared on Etherscan? If not, can you send a screenshot and check your UI and background consoles? 

Try speeding up the gas if it's been pending for a while, all the other transactions have to confirm sequentially so they're blocked by earlier ones.

**Comment by DiMeneses** [comment]
This is the one: https://etherscan.io/tx/0xab74b4b90e02dec2e777db5851d4ce5ff6cbb5ed0e8c7b433b80bbefbe56ff76
(should I not be posting this? I've no idea. Are there any dangers to it?)
All of this occurred when I was trying to get some eggs on Dragonereum... such a shame, I lost my chance to own the first ones

**Comment by tmashuang** [comment]
@DiMeneses gas price is low, consider using https://ethgasstation.info/ for the latest gas prices.

**Comment by DiMeneses** [comment]
Yeah I figured since it just got queued ahahahah I'll try speeding them up. Kinda boring though

**Comment by tmashuang** [comment]
@DiMeneses its 10%, there is works on improving this. You should be able to edit the price manually.

**Comment by ddasbidhar** [comment]
@danfinlay  I also encountered the transactions stuck issue. There are a bunch of pending transactions (both additional transfer and cancellation requests - the queue length is almost 20) behind the original transaction. I can not see the first transaction on etherscan. It shows a message "Sorry, We Are Unable To Locate This transaction Hash" - not able to see the nonce of the stuck transaction. All subsequent transactions show up as "Invalid TxHash  undefined  Entered" on etherscan.  Any help on how to fix it is highly appreciated. 

![image](https://user-images.githubusercontent.com/26621594/49335626-ed8db280-f616-11e8-9305-1404d7cb5fc8.png)

**Comment by LA555** [comment]
Hi there, I have been trying to send transactions from Metamask and have 31 pending transactions in the queue. These include attempts to cancel the transactions and also trying sending different tokens which all have failed. I've also tried restoring Metamask and resending and again these failed. I'm at a complete loss and am not a whiz in all this and it's causing me great distress. I would really appreciate help in accessing my funds and making the transfers. Can anyone help please?

**Comment by bdresser** [comment]
@LA555 please reach out to support@metamask.io and we'll do our best to help/

**Comment by LA555** [comment]
I have already sent an email to support@metamask.io but have received no
response following my last email.

**Comment by DecentralWize** [comment]
It's years later and the same issue is still taking place... Why hasn't MM implemented a way to fix this with an option to add gas for a pending transaction that is missing the "seed up" button on a task? The MEW workaround seems not to be available anymore as well. 

I have multiple tasks stuck in a pending status with no way for me to add gas, that I have been made aware of. Is there still a way to use MyEtherWallet to fix this? Has MM implemented an inhouse fix/workaround over the years of knowing about this complication?

**Comment by dralahce** [comment]
I have the same problem I am really shocked it is possible to have this kind of problems in 2020 and with so innovative projects... it is ridiculous...

**Comment by cabraun** [comment]
Having the same issue. Pending for 10 days. All new transactions remain in Pending. The most recent transactions don't show up on Etherscan. Did anyone figure out how to resolve this? I've tried canceling the first order and 'speeding up' the transaction, but it hasn't resolved the problem.

**Comment by dralahce** [comment]
@cabraun i have been speeding up several times in a row and it finally worked for me. Maybe 10 or 15 times for some of the transactions.

**Comment by cabraun** [comment]
@dralahce Thanks. I speeded up the first transaction about 5-6 times over a couple days and eventually all my pending transactions. Not sure exactly how things were unstuck as I wasn't closely monitoring it. 

Now I have to figure out what all those pending transactions that ultimately went through were.

**Comment by dralahce** [comment]
Great !

**Comment by pravsa** [comment]
@cabraun @dralahce 
Removing MetaMask, re-installing and recovering via seed phrase helped for me! All the pending transactions disappeared and I can send now.

**Comment by Biem101** [comment]
I have the same problem
but my transaction with low gas price is cancelled and is listed in history box of MM.
If i check the hash on etherscan it is still pending. So I cant add to the gas price.

**Comment by pravsa** [comment]
> I have the same problem
> but my transaction with low gas price is cancelled and is listed in history box of MM.
> If i check the hash on etherscan it is still pending. So I cant add to the gas price.

Did you try what I wrote above?

**Comment by Biem101** [comment]
> > I have the same problem
> > but my transaction with low gas price is cancelled and is listed in history box of MM.
> > If i check the hash on etherscan it is still pending. So I cant add to the gas price.
> 
> Did you try what I wrote above?

Yes i did, I deleted MM on the brave browser, and installed it on the chrome browser. Its working again.
thanks a lot!

**Comment by clarupan** [comment]
> @cabraun @dralahce
> Removing MetaMask, re-installing and recovering via seed phrase helped for me! All the pending transactions disappeared and I can send now.

Hey! I tried to do this and my seed phrase is supposedly wrong so i cant get in... any clues?

**Comment by pravsa** [comment]
> > @cabraun @dralahce
> > Removing MetaMask, re-installing and recovering via seed phrase helped for me! All the pending transactions disappeared and I can send now.
> 
> Hey! I tried to do this and my seed phrase is supposedly wrong so i cant get in... any clues?

Sorry, but your seed phrase can't be wrong as it is cryptography and should work with every compatible wallet. If you created the seed with MetaMask it should also work. Seems like you wrote down the wrong words?

**Comment by paradux76** [comment]
Account 

> Having the same issue. Pending for 10 days. All new transactions remain in Pending. The most recent transactions don't show up on Etherscan. Did anyone figure out how to resolve this? I've tried canceling the first order and 'speeding up' the transaction, but it hasn't resolved the problem.



> It's years later and the same issue is still taking place... Why hasn't MM implemented a way to fix this with an option to add gas for a pending transaction that is missing the "seed up" button on a task? The MEW workaround seems not to be available anymore as well.
> 
> I have multiple tasks stuck in a pending status with no way for me to add gas, that I have been made aware of. Is there still a way to use MyEtherWallet to fix this? Has MM implemented an inhouse fix/workaround over the years of knowing about this complication?

Settings > Advanced > Reset Account
"Resetting your account will clear your transaction history. This will not change the balances in your accounts or require you to re-enter your seed phrase."
Removed all 11 of my queued transactions / cancellations.
